{
  if (this.ldapErrorDefinitions == null || this.ldapErrorDefinitions.size() == 0) {
    logger.debug("No error definitions are defined. Throwing error {}",e.getMessage());
    return BadCredentialsAuthenticationException.ERROR;
  }
  logger.debug("Handling error: {}",e.getMessage());
  for (  final LdapErrorDefinition ldapErrorDef : this.ldapErrorDefinitions) {
    if (ldapErrorDef.matches(e.getMessage())) {
      logger.debug("Found error type {}. Throwing error for {}",ldapErrorDef.getType(),e.getMessage());
      return new LdapAuthenticationException(BadCredentialsAuthenticationException.CODE,e.getMessage(),ldapErrorDef.getType());
    }
  }
  logger.debug("No error definition could be matched against the error. Throwing default error for {}",e.getMessage());
  return BadCredentialsAuthenticationException.ERROR;
}
